### PR TITLE
[API] Resolve correct matching of prefixes /admin and /shop

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | api-platform-3|
| Bug fix?        | yes                                                       |
| New feature?    |yes                                                       |
| BC breaks?      | no                                                      |
| Deprecations?   | no|
| Related tickets | |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/ApiPlatform/JsonLd/Serializer/ItemNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Exception\UnexpectedValueException;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Overridden class from the API Platform in order to pass the context to the IRI conversion
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/ApiPlatform/JsonLd/Serializer/ItemNormalizer.php:40]: Could you add comments above the lines that actually changed?
I'd ease keeping this class up to date.

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/providers.xml [hunk 2]
```diff
@@ -15,12 +15,18 @@
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
 >
+    <parameters>
+        <parameter key="sylius.api_path_prefixes" type="collection">
+            <parameter type="constant">Sylius\Bundle\ApiBundle\Provider\PathPrefixes::ADMIN_PREFIX</parameter>
+            <parameter type="constant">Sylius\Bundle\ApiBundle\Provider\PathPrefixes::SHOP_PREFIX</parameter>
+        </parameter>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/providers.xml:22]: Would be nice to have this handled via configuration.
Also, note it in the upgrade file.

**Review by NoResponseMate — COMMENTED** [review]
We're missing the info in the upgrade file on what needs to be set to have sections working correctly.

File: src/Sylius/Bundle/ApiBundle/ApiPlatform/JsonLd/Serializer/ItemNormalizer.php (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/ApiPlatform/JsonLd/Serializer/ItemNormalizer.php:40]: Added 👌🏻

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/providers.xml (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/providers.xml:22]: I've added a note to the UPGRADE file 👌🏻 

About handling these path prefixes via configuration, I would leave the refactor for a separate PR, especially, that I'm not sure if it makes sense here 🤔

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/resolvers.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <defaults public="true" />
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/resolvers.xml]:
```suggestion
        <defaults public="false" />
```

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/resolvers.xml (hunk 3 above)
Comment (TheMilek) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/resolvers.xml]: it is false by default so maybe just remove this line 😅

**Review by TheMilek — APPROVED** [review]

**Comment by TheMilek** [comment]
Thank you @GSadee! 🥇

**Pull request merged**